### PR TITLE
Add hardshrink implementation

### PR DESCRIPTION
**Pull request by mouliraj-mcw**
### Ticket
 Link to Github Issue  #23015

### Problem description
Hardshrink has been implemented as a composite operation.

### What's changed
Added llk support forhardshrink

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist

- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI passes (if applicable)

### CODE REVIEW

File: tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_activations.h [hunk 1]
```diff
@@ -44,13 +45,34 @@ struct ActivationImpl<APPROXIMATION_MODE, ActivationType::Celu>
...
 };
 
+// Specialization for HARDSHRINK activation
+template <bool APPROXIMATION_MODE>
+struct ActivationImpl<APPROXIMATION_MODE, ActivationType::Hardshrink>
+{
+    static inline void apply(sfpi::vFloat& v, uint param0)
+    {
+        float ans = Converter::as_float(param0);
+        v_if (!((v > ans) || (v < -ans)))
```
Comment (amahmudTT) [inline tt_llk_blackhole/common/inc/sfpu/ckernel_sfpu_activations.h:55]: you can just take the abs of v and check if it is less than and (assuming ans is positive). But please check the number of instructions and cycles with the alternative method.

File: tt_llk_wormhole_b0/common/inc/sfpu/ckernel_sfpu_activations.h [hunk 2]
```diff
@@ -44,13 +45,34 @@ struct ActivationImpl<APPROXIMATION_MODE, ActivationType::Celu>
...
 };
 
+// Specialization for HARDSHRINK activation
+template <bool APPROXIMATION_MODE>
+struct ActivationImpl<APPROXIMATION_MODE, ActivationType::Hardshrink>
+{
+    static inline void apply(sfpi::vFloat& v, uint param0)
+    {
+        float ans = Converter::as_float(param0);
+        v_if (!((v > ans) || (v < -ans)))
```
Comment (amahmudTT) [inline tt_llk_wormhole_b0/common/inc/sfpu/ckernel_sfpu_activations.h:55]: Same comment as above.

**Comment by fvranicTT** [comment]
@mouliraj-mcw could you add tests for hardshrink, following the same approach as in celu/gelu? Thanks!

**Comment by amahmudTT** [comment]
Have not been checking this properly, but should this PR pass the post commit tests ?

**Comment by mouliraj-mcw** [comment]
Dropping this PR due to edge case failures in the current approach. I've moved to a compute kernel-based approach instead — please refer to the updated PR here: https://github.com/tenstorrent/tt-metal/pull/24689.
